source/mir/math/stat.d

```diff
@@ -461,6 +443,372 @@ unittest
     assert(float.max.repeat(3).hmean!(double, "fast").approxEqual(float.max));
 }
 
+private
+auto pow(T, U)(in T x, in U power) {
+    static if (is(U : int)) {
+        import mir.math.common: powi;
+        return powi(x, power);
+    } else static if (isFloatingPoint!U && is(T == U)) {
+        import mir.math.common: pow;
+        return pow(x, power);
+    } else {
+        import std.math: pow;
+        return pow(x, power);
```

Comment:
why can't we use `mir.math.common: pow` ?

Reply:
@9il Yes I had realized that wasn't going to work when I was finally able to run the UTs locally, but hadn't gotten around to changing it. I instead created an nthroot function that can also calculate integer roots. I was getting some out of memory errors for uint.max.repeat(3) on my machine, but it works here. I'm happy to adjust the algorithm if you know of a way to reduce the risk of this issue. Or just remove it and keep pow by itself.

Reply:
```
I nthroot(I, J)(in I x, in J n)
    if (isIntegral!(I) && isIntegral!J)
```

How it can return an integer? `gmean` seems should never return an integer, but floating-point number even for integer input. The default FP type can be set to double. The same true for the `median` function.

Reply:
For `gmean`, I was thinking of something like `gmean([3, 9, 64])` that would have an integer return of `12`. 

I was thinking about consistency with other functions in `mir.math.stat`. `mean([1, 2])` will return a integer, I believe `median([1, 2])` will as well. However, this is not true for `hmean([1, 2])` because any integer except `1` should be inversed to a fraction and any fractional input should be a float when passing it in (I think there is a bug in the return type there that I will need to fix). 

The only way to guarantee that `mean([1, 2])` always produces `1.5` is to force the user to put a type in, as in `mean!float([1, 2])`, and probably disallow mean!int([1, 2]). I have no issue doing this, but it wasn't what we had done for the others.

Reply:
I am sorry for ignoring this issue before. Let's go with the FP return type for all similar functions, or just for gmean for this PR and fix others later.

Reply:
No problem. 

I think it will be important to still allow something like `mean([1, 2])` (and maybe  even `mean!int([1, 2])`) but have the result be a floating point type. The problem is that if you try to just prevent the use of integral types, such as by placing template constraints on `F` with `isFloatingPoint!F`, then you also can no longer call it with user-defined types or complex types. However, I can instead adjust `gmeanType` with special handling for when it is not a floating point type.

Reply:
Hmm, this may be more of an issue for `std.traits.isFloatingPoint` than the mir version.